include/seastar/http/request.hh

```diff
@@ -76,6 +79,22 @@ struct request {
     noncopyable_function<future<>(output_stream<char>&&)> body_writer; // for client
     int listener_idx;
 
+    /**
+     * Get the address of the client that generated the request
+     * @return The address of the client that generated the request
+     */
+    const socket_address & get_client_address() const {
+        return _client_address;
+    }
+
+    /**
+     * Get the address of the server that handled the request
+     * @return The address of the server that handled the request
+     */
+    const socket_address & get_server_address() const {
+        return _server_address;
```

Comment:
wouldn't our server already have this information available? it might not already be plumbed into request handlers, but in principle it's established when the server is initialized and request handler could get it through captured reference to the server?

Reply:
We know what addresses we use for the listeners but if a listener is listening on`0.0.0.0` it could come in via any network interface.

Reply:
ahh